fix: fix VR

FlayaN opened the pull request:
Fixed the offsets I could find, also added a VR check if the user have the correct vr-address-library version

These offsets have all been verified in IDA

~Draft until vr-address-library is updated with the new id `524768`~ Added in vr-address-library 0.140.0

closes #331 

Added VR support to these features:
- Cloud shadows
- Skylighting
- Terrain Blending

FlayaN committed:
fix: fix VR offsets

alandtse commented:
Should be just a fix not feat.

alandtse commented on src/Deferred.cpp:
```diff
@@ -524,7 +524,7 @@ void Deferred::EndDeferred()
 	}
 
 	for (uint i = 4; i < 8; i++) {
-		shadowState->GetRuntimeData().renderTargets[i] = RE::RENDER_TARGET::kNONE;
```
Why is this given the GET_INSTANCE_MEMBER macro?

alandtse replied:
Oh nevermind, I'm reading the diff backwards.

FlayaN commented on src/XSEPlugin.cpp:
```diff
@@ -148,6 +148,10 @@ bool Load()
 	}
 
+	if (REL::Module::IsVR()) {
+		REL::IDDatabase::get().IsVRAddressLibraryAtLeastVersion("0.138.0", true);
```
bump "0.138.0" to "0.140.0" (the actual version that is required)

alandtse committed:
fix: prevent extra init of State

alandtse committed:
fix: fix enumeration of renderTargets

Total will change based on Skyrim version.

FlayaN commented on extern/CommonLibSSE-NG:
Maybe update to the latest?

alandtse replied:
Don't do it yet. Had some variable renames to sync with the rest of commonlib world naming. My local branch of CS fixes the renames.

alandtse committed:
chore: add more info in vrs logs

FlayaN commented:
Getting this error while compiling shaders:
`Data\Shaders\Lighting.hlsl(1507,19-91): error X3004: undeclared identifier 'GetSkylightOcclusion'`

FlayaN committed:
fix: add SKYLIGHTING defintion check around GetSkylightOcclusion

FlayaN committed:
fix: bump vr-address-library requirement to 0.140.0

FlayaN committed:
fix: fix debug view in deferred VR

FlayaN commented:
Might have a fix for the "everything being dark issue"

FlayaN committed:
fix: deferred rendering ambient issue

alandtse committed:
feat(VR): enable skylighting

alandtse committed:
fix: fix sky compilation error in VR

alandtse committed:
fix: fix water rendering

alandtse committed:
fix(VR): fix screen space shadows

Fix loading of ScreenSpaceShadowsTexture stereo buffer using a non
stereo uv coordinate.

alandtse committed:
fix: fix bad uv conversion

Typically dtid is in stereo so needs to be corrected to nonstereo.

alandtse committed:
feat(VR): enable sky and cloud shadows

alandtse committed:
feat(VR): enable Terrain Blending

alandtse committed:
fix: fix flat compilation error

FlayaN committed:
refactor: create util for eyePos and cameraData

doodlum committed:
fix: incorrect reflections

alandtse committed:
feat: add particle shader

alandtse committed:
feat: add bloodsplatter shader

VR and Flat look identical in RE. With this we have parity across both.
